coala_quickstart/green_mode/green_mode_core.py

```diff
@@ -0,0 +1,80 @@
+import os
+
+from coala_quickstart.green_mode.green_mode import (
+    initialize_project_data, dump_to_file, get_contents,
+    run_quickstartbear, bear_test_fun)
+from coala_quickstart.green_mode.filename_operations import (
+    check_filename_prefix_postfix)
+
+
+def green_mode(project_dir, ignore_globs, bears, bear_settings_obj,
+               op_args_limit, value_to_op_args_limit, printer=None):
+    """
+    Runs the green mode of coala-quickstart.
+
+    Generates '.project_data.yaml' which contains the files and directory
+    structure of the project, runs the QuickstartBear which guesses some values
+    of settings the can take an infinite set of values by parsing the
+    file_dict and appends to `.project_data.yaml`. Runs some further linting
+    options based on file names etc. Calls the methods which test out whether
+    a setting value is green for a bear i.e. does not point out any error in
+    the code base and further generates sections and writes the green config
+    file for the project.
+    :param project_dir:
+        The project directory.
+    :param ignore_globs:
+        The globs of the files to ignore from the linting process.
+    :param bears:
+        The bears from Constants/GREEN_MODE_COMPATIBLE_BEAR_LIST along
+        with Constants/IMPORTANT_BEAR_LIST.
+    :param bear_settings_obj:
+        The object of SettingsClass/BearSettings which stores the metadata
+        about whether a setting takes a boolean value or any other value.
+    :param op_args_limit:
+        The maximum number of optional bear arguments allowed for guessing.
+    :param value_to_op_args_limit:
+        The maximum number of values to run the bear again and again for
+        a optioanl setting.
+    """
+    from coala_quickstart.green_mode.filename_operations import (
+        check_filename_prefix_postfix)
+    ignore_globs.append(project_dir+os.sep+'.git'+os.sep+'**')
```

Comment:
Add spaces around `+`

Reply:
or you can use `os.path.join` for this which will take care of separators itself.

Reply:
^^ same for elsewhere in the file

Reply:
Just figured out `os.sep` is faster, but since we are using it only once or twice, we should go with `os.path.join` which is more readable and clean. https://stackoverflow.com/questions/16789714/differences-between-use-of-os-path-join-and-os-sep-concatenation